mlir-sycl/lib/Dialect/IR/SYCLOpsDialect.cpp

```diff
@@ -66,18 +71,48 @@ class SYCLInlinerInterface : public mlir::DialectInlinerInterface {
   }
 };
 
+class SYCLOpAsmInterface final : public mlir::OpAsmDialectInterface {
+public:
+  using mlir::OpAsmDialectInterface::OpAsmDialectInterface;
+
+public:
+  AliasResult getAlias(mlir::Type Type, llvm::raw_ostream &OS) const final;
+};
+
+mlir::OpAsmDialectInterface::AliasResult
+SYCLOpAsmInterface::getAlias(mlir::Type Type, llvm::raw_ostream &OS) const {
+  return llvm::TypeSwitch<mlir::Type, AliasResult>(Type)
+      .Case<mlir::sycl::AccessorType, mlir::sycl::ItemType,
+            mlir::sycl::ItemBaseType>([&](auto Ty) {
+        OS << "sycl_" << decltype(Ty)::getMnemonic() << "_" << Ty.getDimension()
+           << "_";
```

Comment:
Why do we want to add `_` at the end?

Reply:
From [this interface definition](https://github.com/intel/llvm/blob/ffef38bb339190a24fb62ad1aacdc11f39677e07/mlir/include/mlir/IR/OpImplementation.h#L1564):
```cpp
/// Hooks for getting an alias identifier alias for a given symbol, that is
/// not necessarily a part of this dialect. The identifier is used in place of
/// the symbol when printing textual IR. These aliases must not contain `.` or
/// end with a numeric digit([0-9]+).
```

Reply:
Too bad ...